oap-server/server-receiver-plugin/skywalking-ebpf-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/ebpf/provider/handler/AccessLogServiceHandler.java

```diff
@@ -0,0 +1,700 @@
+import org.apache.skywalking.oap.server.core.config.NamingControl;
+import org.apache.skywalking.oap.server.core.source.K8SEndpoint;
+import org.apache.skywalking.oap.server.core.source.K8SEndpointRelation;
+import org.apache.skywalking.oap.server.core.source.K8SMetrics;
+import org.apache.skywalking.oap.server.core.source.K8SService;
+import org.apache.skywalking.oap.server.core.source.K8SServiceInstance;
+import org.apache.skywalking.oap.server.core.source.K8SServiceInstanceRelation;
+import org.apache.skywalking.oap.server.core.source.K8SServiceRelation;
+import org.apache.skywalking.oap.server.core.source.SourceReceiver;
+import org.apache.skywalking.oap.server.library.module.ModuleManager;
+import org.apache.skywalking.oap.server.library.util.StringUtil;
+import org.apache.skywalking.oap.server.telemetry.TelemetryModule;
+import org.apache.skywalking.oap.server.telemetry.api.CounterMetrics;
+import org.apache.skywalking.oap.server.telemetry.api.HistogramMetrics;
+import org.apache.skywalking.oap.server.telemetry.api.MetricsCreator;
+import org.apache.skywalking.oap.server.telemetry.api.MetricsTag;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Slf4j
+public class AccessLogServiceHandler extends EBPFAccessLogServiceGrpc.EBPFAccessLogServiceImplBase {
+    private final SourceReceiver sourceReceiver;
+    private final NamingControl namingControl;
+
+    private final CounterMetrics inCounter;
+    private final HistogramMetrics processHistogram;
+    private final CounterMetrics dropCounter;
+    private final ConcurrentHashMap<String, DropDataReason> dropReasons = new ConcurrentHashMap<>();
+
+    public AccessLogServiceHandler(ModuleManager moduleManager) {
+        this.sourceReceiver = moduleManager.find(CoreModule.NAME).provider().getService(SourceReceiver.class);
+        this.namingControl = moduleManager.find(CoreModule.NAME).provider().getService(NamingControl.class);
+
+        MetricsCreator metricsCreator = moduleManager.find(TelemetryModule.NAME)
+            .provider()
+            .getService(MetricsCreator.class);
+        this.inCounter = metricsCreator.createCounter(
+            "k8s_als_in_count", "The count of envoy ALS message received", MetricsTag.EMPTY_KEY,
+            MetricsTag.EMPTY_VALUE);
+        this.processHistogram = metricsCreator.createHistogramMetric(
+            "k8s_als_in_latency", "The process latency of envoy ALS message received", MetricsTag.EMPTY_KEY,
+            MetricsTag.EMPTY_VALUE);
+        this.dropCounter = metricsCreator.createCounter(
+            "k8s_als_drop_count", "The count of envoy ALS message dropped", MetricsTag.EMPTY_KEY,
+            MetricsTag.EMPTY_VALUE);
```

Comment:
As these important self-observability metrics, please update the native self-observability dashboard and Grafana dashboard, we should make them side by side with Envoy ALS and Agent traces metrics.

Reply:
Update, please take a look at the screenshots.